Hmm tests are failing:

```
 INFO  [alembic.runtime.migration] Running upgrade 7b2661a43ba3 -> be2bfac3da23, Add has_import_errors column to DagModel
  WARNI [airflow.models.crypto] empty cryptography key - values will not be stored encrypted.
  Traceback (most recent call last):
    File "/usr/local/lib/python3.7/site-packages/sqlalchemy/engine/base.py", line 1277, in _execute_context
      cursor, statement, parameters, context
    File "/usr/local/lib/python3.7/site-packages/sqlalchemy/engine/default.py", line 608, in do_execute
      cursor.execute(statement, parameters)
  psycopg2.errors.LockNotAvailable: canceling statement due to lock timeout
  
  
  The above exception was the direct cause of the following exception:
  
  Traceback (most recent call last):
    File "/opt/airflow/airflow/utils/db.py", line 1017, in create_global_lock
      conn.execute(text('SELECT pg_advisory_lock(:id)'), id=lock.value)
    File "/usr/local/lib/python3.7/site-packages/sqlalchemy/engine/base.py", line 1011, in execute
      return meth(self, multiparams, params)
    File "/usr/local/lib/python3.7/site-packages/sqlalchemy/sql/elements.py", line 298, in _execute_on_connection
      return connection._execute_clauseelement(self, multiparams, params)
    File "/usr/local/lib/python3.7/site-packages/sqlalchemy/engine/base.py", line 1130, in _execute_clauseelement
      distilled_params,
    File "/usr/local/lib/python3.7/site-packages/sqlalchemy/engine/base.py", line 1317, in _execute_context
      e, statement, parameters, cursor, context
    File "/usr/local/lib/python3.7/site-packages/sqlalchemy/engine/base.py", line 1511, in _handle_dbapi_exception
      sqlalchemy_exception, with_traceback=exc_info[2], from_=e
    File "/usr/local/lib/python3.7/site-packages/sqlalchemy/util/compat.py", line 182, in raise_
      raise exception
    File "/usr/local/lib/python3.7/site-packages/sqlalchemy/engine/base.py", line 1277, in _execute_context
      cursor, statement, parameters, context
    File "/usr/local/lib/python3.7/site-packages/sqlalchemy/engine/default.py", line 608, in do_execute
      cursor.execute(statement, parameters)
  sqlalchemy.exc.OperationalError: (psycopg2.errors.LockNotAvailable) canceling statement due to lock timeout
  
  [SQL: SELECT pg_advisory_lock(%(id)s)]
  [parameters: {'id': 1}]
  (Background on this error at: http://sqlalche.me/e/13/e3q8)
  
  During handling of the above exception, another exception occurred:
  
  Traceback (most recent call last):
    File "/usr/local/lib/python3.7/site-packages/sqlalchemy/engine/base.py", line 1277, in _execute_context
      cursor, statement, parameters, context
    File "/usr/local/lib/python3.7/site-packages/sqlalchemy/engine/default.py", line 608, in do_execute
      cursor.execute(statement, parameters)
  psycopg2.errors.InFailedSqlTransaction: current transaction is aborted, commands ignored until end of transaction block
  
  
  The above exception was the direct cause of the following exception:
  
  Traceback (most recent call last):
    File "/usr/local/bin/airflow", line 33, in <module>
      sys.exit(load_entry_point('apache-airflow', 'console_scripts', 'airflow')())
    File "/opt/airflow/airflow/__main__.py", line 48, in main
      args.func(args)
    File "/opt/airflow/airflow/cli/cli_parser.py", line 48, in command
      return func(*args, **kwargs)
    File "/opt/airflow/airflow/cli/commands/db_command.py", line 39, in resetdb
      db.resetdb()
    File "/opt/airflow/airflow/utils/session.py", line 69, in wrapper
      return func(*args, session=session, **kwargs)
    File "/opt/airflow/airflow/utils/db.py", line 932, in resetdb
      initdb(session=session)
    File "/opt/airflow/airflow/utils/session.py", line 66, in wrapper
      return func(*args, **kwargs)
    File "/opt/airflow/airflow/utils/db.py", line 599, in initdb
      with create_global_lock(session=session, lock=DBLocks.MIGRATIONS):
    File "/usr/local/lib/python3.7/contextlib.py", line 112, in __enter__
      return next(self.gen)
    File "/opt/airflow/airflow/utils/db.py", line 1030, in create_global_lock
      conn.execute('SET LOCK_TIMEOUT TO DEFAULT')
    File "/usr/local/lib/python3.7/site-packages/sqlalchemy/engine/base.py", line 1003, in execute
      return self._execute_text(object_, multiparams, params)
    File "/usr/local/lib/python3.7/site-packages/sqlalchemy/engine/base.py", line 1178, in _execute_text
      parameters,
    File "/usr/local/lib/python3.7/site-packages/sqlalchemy/engine/base.py", line 1317, in _execute_context
      e, statement, parameters, cursor, context
    File "/usr/local/lib/python3.7/site-packages/sqlalchemy/engine/base.py", line 1511, in _handle_dbapi_exception
      sqlalchemy_exception, with_traceback=exc_info[2], from_=e
    File "/usr/local/lib/python3.7/site-packages/sqlalchemy/util/compat.py", line 182, in raise_
      raise exception
    File "/usr/local/lib/python3.7/site-packages/sqlalchemy/engine/base.py", line 1277, in _execute_context
      cursor, statement, parameters, context
    File "/usr/local/lib/python3.7/site-packages/sqlalchemy/engine/default.py", line 608, in do_execute
      cursor.execute(statement, parameters)
  sqlalchemy.exc.InternalError: (psycopg2.errors.InFailedSqlTransaction) current transaction is aborted, commands ignored until end of transaction block
  
  [SQL: SET LOCK_TIMEOUT TO DEFAULT]
  (Background on this error at: http://sqlalche.me/e/13/2j85)
```

--- a/airflow/utils/db.py
+++ b/airflow/utils/db.py
@@ -995,8 +995,8 @@ class DBLocks(enum.IntEnum):
     """
     Cross-db Identifiers for advisory global database locks.
 
-    Postgres uses int64 lock ids so we use the integer value, MySQL uses names, so we use the ``_name_``
-    field.
+    Postgres uses int64 lock ids so we use the integer value, MySQL uses names, so we
+    call ``str()`, which is implemented using the ``_name_`` field.
     """
 
     MIGRATIONS = enum.auto()
@@ -1018,7 +1018,7 @@ def create_global_lock(session, lock: DBLocks, lock_timeout=1800):
         elif dialect.name == 'mysql' and dialect.server_version_info >= (5, 6):
             conn.execute(text("SELECT GET_LOCK(:id, :timeout)"), id=str(lock), timeout=lock_timeout)
         elif dialect.name == 'mssql':
-            # TODO: make locking works for MSSQL
+            # TODO: make locking work for MSSQL
             pass
 
         yield
@@ -1031,5 +1031,5 @@ def create_global_lock(session, lock: DBLocks, lock_timeout=1800):
         elif dialect.name == 'mysql' and dialect.server_version_info >= (5, 6):
             conn.execute(text("select RELEASE_LOCK(:id)"), id=str(lock))
         elif dialect.name == 'mssql':
-            # TODO: make locking works for MSSQL
+            # TODO: make locking work for MSSQL
             pass

--- a/airflow/utils/db.py
+++ b/airflow/utils/db.py
@@ -1009,7 +1009,7 @@ class DBLocks(enum.IntEnum):
 @contextlib.contextmanager
 def create_global_lock(session, lock: DBLocks, lock_timeout=1800):
     """Contextmanager that will create and teardown a global db lock."""
-    conn = session.connection()
+    conn = session.get_bind().connect()
     dialect = conn.dialect
     try:
         if dialect.name == 'postgresql':
@@ -1021,14 +1021,13 @@ def create_global_lock(session, lock: DBLocks, lock_timeout=1800):
             # TODO: make locking works for MSSQL
             pass
 
-        yield None
+        yield
     finally:
-        # The session may have been "closed" (which is fine, the lock lasts more than a transaction) -- ensure
-        # we get a usable connection
-        conn = session.connection()
         if dialect.name == 'postgresql':
             conn.execute('SET LOCK_TIMEOUT TO DEFAULT')
-            conn.execute(text('SELECT pg_advisory_unlock(:id)'), id=lock.value)
+            (unlocked,) = conn.execute(text('SELECT pg_advisory_unlock(:id)'), id=lock.value).fetchone()
+            if not unlocked:
+                raise RuntimeError("Error releasing DB lock!")
         elif dialect.name == 'mysql' and dialect.server_version_info >= (5, 6):
             conn.execute(text("select RELEASE_LOCK(:id)"), id=str(lock))
         elif dialect.name == 'mssql':

